packages/1155-contracts/package/premint-api.test.ts

```diff
@@ -0,0 +1,208 @@
+        {
+          chain_name: "ZORA-TESTNET",
+          collection: {
+            contractAdmin: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+            contractName: "Testing Contract",
+            contractURI: "https://zora.co/testing/contract.json",
+          },
+          premint: {
+            deleted: false,
+            tokenConfig: {
+              fixedPriceMinter: "0x04E2516A2c207E84a1839755675dfd8eF6302F0a",
+              maxSupply: "18446744073709551615",
+              maxTokensPerAddress: "0",
+              mintDuration: "604800",
+              mintStart: "0",
+              pricePerToken: "18446744073709551615",
+              royaltyBPS: 1000,
+              royaltyMintSchedule: 0,
+              royaltyRecipient: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+              tokenURI: "https://zora.co/testing/token.json",
+            },
+            uid: 3,
+            version: 1,
+          },
+          signature:
+            "0x1560c8bd247c7432c1e53049fd0ccb1287df6e7cad4649718417ed7f08db45c87f6ba03aafb61dcbfbbd66fafddaa63bbe303bb9bcd7abd30ebc49becda5952f1c",
+        }
+      );
+
+      console.log({ premint });
+    },
+    20 * 1000
+  );
+
+  it<TestContext>(
+    "can execute premint on network",
+    async ({ fixedPriceMinterAddress, forkedChainId, anvilChainId }) => {
+      const preminterApi = new PreminterAPI(foundry);
+
+      preminterApi.get = vi.fn().mockResolvedValue({
+        chain_name: "ZORA-TESTNET",
+        collection: {
+          contractAdmin: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+          contractName: "Testing Contract",
+          contractURI: "https://zora.co/testing/contract.json",
+        },
+        premint: {
+          deleted: false,
+          tokenConfig: {
+            fixedPriceMinter: "0x04E2516A2c207E84a1839755675dfd8eF6302F0a",
+            maxSupply: "18446744073709551615",
+            maxTokensPerAddress: "0",
+            mintDuration: "604800",
+            mintStart: "0",
+            pricePerToken: "18446744073709551615",
+            royaltyBPS: 1000,
+            royaltyMintSchedule: 0,
+            royaltyRecipient: "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+            tokenURI: "https://zora.co/testing/token.json",
+          },
+          uid: 3,
+          version: 1,
+        },
+        signature:
+          "0x1560c8bd247c7432c1e53049fd0ccb1287df6e7cad4649718417ed7f08db45c87f6ba03aafb61dcbfbbd66fafddaa63bbe303bb9bcd7abd30ebc49becda5952f1c",
+      });
+      preminterApi.post = vi.fn();
+
+      console.log({deployerAccount})
+
+      const premint = await preminterApi.executePremintWithWallet({
+        data: await preminterApi.getPremintData('0x0bdD2Fcb03912403c0B4699EDBB6bDAd65dACf62', 3),
+        account: deployerAccount,
+        walletClient,
+        publicClient,
+        mintArguments: {
+          quantityToMint: 1n,
+          mintComment: '',
+        },
+      });
+
+      console.log({ premint });
```

Comment:
how about a final expectation? like checking the status with an rpc read on the premint contract?